ui/hooks/useCarouselManagement/useCarouselManagement.ts

```diff
@@ -87,20 +143,32 @@ export const useCarouselManagement = ({
         ...SWEEPSTAKES_SLIDE,
         dismissed: false,
       };
-      defaultSlides.unshift(newSweepstakesSlide);
-    } else if (existingSweepstakesSlide?.dismissed) {
+      baseSlides.unshift(newSweepstakesSlide);
+    } else if (isSweepstakesSlideDismissed) {
       // Add the sweepstakes slide with the dismissed state preserved
       // We need this to maintain the persisted dismissed state
       const dismissedSweepstakesSlide = {
         ...SWEEPSTAKES_SLIDE,
         dismissed: true,
       };
 
-      defaultSlides.push(dismissedSweepstakesSlide);
+      baseSlides.push(dismissedSweepstakesSlide);
     }
 
-    dispatch(updateSlides(defaultSlides));
-  }, [dispatch, hasZeroBalance, isRemoteModeEnabled, slides, testDate, inTest]);
+    return baseSlides;
+  }, [totalBalance, isRemoteModeEnabled, isSweepstakesActive, slides, inTest]);
+
+  const lastPayload = useRef<CarouselSlide[] | null>(null);
+  useEffect(() => {
+    // finally, and *most importantly*, dispatch only when the payload is
+    // *actually* different from the last one. This is important to avoid
+    // unnecessary calls to `updateSlides`, which will cause the carousel to
+    // re-render
+    if (!isEqual(lastPayload.current, defaultSlides)) {
+      dispatch(updateSlides(defaultSlides));
+      lastPayload.current = defaultSlides;
+    }
+  }, [dispatch, defaultSlides]);
```

Comment:
One of the big differences here is that the dispatch and the computation of defaultSlides have different dependencies (note that: defaultSlides is memoized now).

The other is the use of isEqual to really check if anything has changed.